### PR TITLE
[synthetics-command] Do not try to open tunnel if there are no test to run

### PR DESCRIPTION
**Pull request by KevinYeramian**

### CODE REVIEW

File: src/commands/synthetics/run-test.ts [hunk 1]
```diff
@@ -167,13 +174,6 @@ export class RunTestCommand extends Command {
       return safeExit(1)
     }
 
-    // All tests have been skipped or are missing.
-    if (!tests.length) {
-      this.reporter.log('No test to run.\n')
-
-      return safeExit(0)
-    }
```
Comment (etnbrd) [inline src/commands/synthetics/run-test.ts:175]: What was the point of reporting this after trying to run the tests?

File: src/commands/synthetics/run-test.ts [hunk 2]
```diff
@@ -119,6 +119,13 @@ export class RunTestCommand extends Command {
 
     const publicIdsToTrigger = tests.map(({public_id}) => public_id)
 
+    // All tests have been skipped or are missing.
+    if (!tests.length) {
+      this.reporter.log('No test to run.\n')
+
+      return safeExit(0)
+    }
+
```
[the remark this answers is not in the document]
Comment (m-rousse) [reply src/commands/synthetics/run-test.ts:128]: `overridenTestsToTrigger` contains all test configs, including skipped ones. The only reason a test is not included in `overridenTestsToTrigger` is when it has not been found or the backend returned a 5xx (and the `--failOnCriticalErrors` flag is set to false, else it crashes directly)
This safeguard also takes skipped tests into account.

**Review by m-rousse — APPROVED** [review]
As a nit, you may move the safe guard before the declaration of `publicIdsToTrigger`.
Else, all good.

File: src/commands/synthetics/run-test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (etnbrd) [reply src/commands/synthetics/run-test.ts:126]: Out of curiosity, as it's too late for this PR anyway, isn't it odd to have two very similar safe-guard at completely different places in the code-base? Wouldn't it make sense to bring them closer?

**Pull request merged**